pkg/controllers/management/auth/auth_config.go

```diff
@@ -33,29 +34,46 @@ type CleanupService interface {
 }
 
 type authConfigController struct {
-	users            v3.UserLister
-	authRefresher    providerrefresh.UserAuthRefresher
-	cleanup          CleanupService
-	authConfigClient controllers.AuthConfigClient
+	users         v3.UserLister
+	authRefresher providerrefresh.UserAuthRefresher
+	cleanup       CleanupService
+	authConfigs   objectclient.GenericClient
 }
 
 func newAuthConfigController(context context.Context, mgmt *config.ManagementContext, scaledContext *config.ScaledContext) *authConfigController {
 	controller := &authConfigController{
-		users:            mgmt.Management.Users("").Controller().Lister(),
-		authRefresher:    providerrefresh.NewUserAuthRefresher(context, scaledContext),
-		cleanup:          cleanup.NewCleanupService(mgmt.Core.Secrets(""), mgmt.Wrangler.Mgmt),
-		authConfigClient: mgmt.Wrangler.Mgmt.AuthConfig(),
+		users:         mgmt.Management.Users("").Controller().Lister(),
+		authRefresher: providerrefresh.NewUserAuthRefresher(context, scaledContext),
+		cleanup:       cleanup.NewCleanupService(mgmt.Core.Secrets(""), mgmt.Wrangler.Mgmt),
+		authConfigs:   scaledContext.Management.AuthConfigs("").ObjectClient().UnstructuredClient(),
 	}
 	return controller
 }
 
 func (ac *authConfigController) setCleanupAnnotation(obj *v3.AuthConfig, value string) (runtime.Object, error) {
-	if obj.Annotations == nil {
-		obj.Annotations = make(map[string]string)
+	name := obj.Name
+	o, err := ac.authConfigs.Get(name, v1.GetOptions{})
+	if err != nil {
+		return nil, err
 	}
-	objCopy := obj.DeepCopy()
-	objCopy.Annotations[CleanupAnnotation] = value
-	return ac.authConfigClient.Update(objCopy)
+	u, ok := o.(runtime.Unstructured)
+	if !ok {
+		logrus.Errorf("auth config %s is not an unstructured value", name)
+		return nil, nil
+	}
+	authConfig := u.UnstructuredContent()
+	metadata, ok := authConfig["metadata"].(map[string]any)
+	if !ok {
+		logrus.Errorf("auth config %s does not have the 'metadata' field", name)
+		return nil, nil
+	}
+	annotations, ok := metadata["annotations"].(map[string]any)
+	if !ok {
+		annotations = make(map[string]any)
+	}
+	annotations[CleanupAnnotation] = value
```

Comment:
I think that you are relying on having existing annotations for this to work. If you hit the empty case and need to make a new map, this won't update the annotations on the authconfig, since `annotations` aren't stored anywhere on the auth config.

You can either:

- Add a line after this one to update `authConfig["metadata"]["annotations"]` to the new annotations (covering both the "no existing annotations" and "existing annotations" cases) - I prefer this option.
- Add a line in the "no existing annotations" case to update `authconfig["metadata"]["annotations"]` to the new empty value